deep_qa/data/instances/text_classification/frame_instance.py

```diff
@@ -0,0 +1,198 @@
+from typing import Dict, List
+
+import numpy
+from overrides import overrides
+
+from ..instance import TextInstance, IndexedInstance
+from ...data_indexer import DataIndexer
+
+# TODO PR request for having these in the json as an application specific content
+# the slotnames can vary according to different end applications, e.g., a HowTo tuple, OpenIE tuple ...
+SLOTNAMES_ORDERED = ["agent", "beneficiary", "causer", "context", "definition", "event",
```

Comment:
If you want to always have a fixed set of slots that every dataset uses, where your input includes a one-hot vector for which slot you're trying to fill, or something like that, this should work.  The model's weights will be tied to particular slots, so a model trained on one slot configuration will not possibly generalize to any other slot configuration.

If you want to make this more modular, you could consider using a separate namespace in the data indexer to keep track of this for you, instead of hard-coding it here.  Something similar is done with tag labels [here](https://github.com/allenai/deep_qa/blob/dd6f20cd0fde902cc54f58baff29f355a356c3cd/deep_qa/data/instances/sequence_tagging/tagging_instance.py#L30).